tfimm/train/train.py

```diff
@@ -80,8 +80,9 @@ def run(cfg: ExperimentConfig, parse_args: bool = False):
 
     # Construct datasets
     train_ds = get_class(cfg.train_dataset_class)(cfg=cfg.train_dataset)
-    # TODO: Make val dataset optional
-    val_ds = get_class(cfg.val_dataset_class)(cfg=cfg.val_dataset)
+    val_ds = None
+    if cfg.val_dataset and cfg.val_dataset_class:
```

Comment:
I would make the check here `if cfg.val_dataset_class:`, the assumption being, if we have a value for `val_dataset_class` in `cfg`, we expect correct config to be present in the key `val_dataset`. We don't want to silently skip validation if `val_dataset` doesn't contain a valid config.